src/ServiceControl/SagaAudit/SagaAuditing.cs

```diff
@@ -35,11 +35,26 @@ public override void Enrich(IReadOnlyDictionary<string, string> headers, IDictio
 
                         foreach (var part in multiSagaChanges.Select(s => s.Split(':')))
                         {
-                            sagasChanges.Add(part[0], part[1]);
+                            var id = part[0];
+                            var thisChange = part[1];
+                            string previousChange;
+                            if (!sagasChanges.TryGetValue(id, out previousChange))
+                            {
+                                sagasChanges[id] = thisChange;
+                            }
+                            else //Completed overrides New overrides Updated
+                            {
+                                if (string.Equals(thisChange, "Completed", StringComparison.OrdinalIgnoreCase)
+                                    || string.Equals(thisChange, "New", StringComparison.OrdinalIgnoreCase) && string.Equals(previousChange, "Updated", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    sagasChanges[id] = thisChange;
+                                }
+                            }
                         }
                     }
 
                     var sagas = sagasInvokedRaw.Split(';')
```

Comment:
Urgh. We might need to create a separate Parser to pull those apart. Are the Ids always going to be Guids? At least then we know the length of them.